crates/conjure_core/src/rule_engine/rule.rs

```diff
@@ -72,25 +76,25 @@ impl Reduction {
     }
 
     /// Represents a reduction with no side effects on the model.
-    pub fn pure(new_expression: Expression) -> Self {
+    pub fn pure(new_expression: Vec<Expression>) -> Self {
         Self {
             new_expression,
-            new_top: Expression::Nothing,
+            new_top: Vec::new(),
             symbols: SymbolTable::new(),
         }
     }
 
     /// Represents a reduction that also modifies the symbol table.
-    pub fn with_symbols(new_expression: Expression, symbols: SymbolTable) -> Self {
+    pub fn with_symbols(new_expression: Vec<Expression>, symbols: SymbolTable) -> Self {
```

Comment:
propagate that change here too. we just want to change the constraints in the model (and new_top as well) to be a vec, not everything else.